src/app/rundown-view/components/follow-playhead-timeline/follow-playhead-timeline.component.ts

```diff
@@ -86,16 +86,16 @@ export class FollowPlayheadTimelineComponent implements OnChanges {
     }
   }
 
-  private checkForSpanningElements(parts: readonly Part[]): Part[] {
-    return parts.reduce((result: Part[], currentPart: Part, index: number) => {
-      const previousPart = result[index - 1]
+  private augmentPartsWithSpanningPiece(parts: readonly Part[]): Part[] {
+    return parts.map((part, index, allParts) => {
+      const previousPart = allParts[index - 1]
       const prevPieceWithSpanElement = previousPart?.pieces.findIndex(piece => piece.lifespan === PieceLifespan.SPANNING_UNTIL_SEGMENT_END)
```

Comment:
A very minor and subjective opinion here is that we don't usually abbreviate words, so i'd write `previous` and `current` everywhere even though it might be long names. Though i do understand that it is very common :-D

Reply:
Gotcha! Will change this then :)